src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,187 @@
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { fetchPeopleData } from '../utils/fetchPeopleData';
+import { SortOrder, SortQuery } from '../types/SortParams';
+
+function calcCentury(year: number) {
+  return Math.ceil(year / 100);
+}
+
+const {
+  NAME, SEX, BORN, DIED, INITIAL,
+} = SortQuery;
+
+const { ASC } = SortOrder;
 
 export const PeoplePage = () => {
+  const [peopleData, setPeopleData] = useState<Person[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const [filteredList, setFilteredList] = useState<Person[] | null>(null);
+
+  const location = useLocation();
+  const initialSearchParams = new URLSearchParams(location.search);
+  const initialSearchQuery = initialSearchParams.get('searchQuery') || '';
+  const initialSexParam = initialSearchParams.get('sex') || '';
+  const initialCenturyParams = initialSearchParams.getAll('century');
+
+  const initialSortQuery = initialSearchParams.get('sort') || SortQuery.INITIAL;
+  const initialOrderQuery = initialSearchParams.get('order') || SortOrder.ASC;
+
+  const [searchQuery, setSearchQuery] = useState(initialSearchQuery);
+  const [sex, setSex] = useState(initialSexParam);
+  const [centuries, setCenturies] = useState(initialCenturyParams);
+
+  const [sortQuery, setSortQuery] = useState<SortQuery>(
+    initialSortQuery as SortQuery,
+  );
+  const [sortOrder, setSortOrder] = useState<SortOrder>(
+    initialOrderQuery as SortOrder,
+  );
+  const [sortedList, setSortedList] = useState<Person[]>([]);
+
+  const initialList = useMemo(() => {
+    if (filteredList) {
+      return [...filteredList];
+    }
+
+    return [];
+  }, [filteredList]);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    fetchPeopleData()
+      .then((data) => {
+        setPeopleData(data);
+      })
+      .catch(() => {
+        setIsLoading(false);
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (!peopleData) {
+      return;
+    }
+
+    const filteredData = peopleData.filter((person) => {
+      const nameMatch
+        = initialSearchQuery === ''
+        || person.name.toLowerCase().includes(initialSearchQuery)
+        || person.motherName?.toLowerCase().includes(initialSearchQuery)
+        || person.fatherName?.toLowerCase().includes(initialSearchQuery);
+
+      const sexMatch = initialSexParam === '' || person.sex === initialSexParam;
+
+      const centuryMatch
+        = initialCenturyParams.length === 0
+        || initialCenturyParams.includes(calcCentury(person.born).toString())
+        || initialCenturyParams.includes(calcCentury(person.died).toString());
+
+      return nameMatch && sexMatch && centuryMatch;
+    });
+
+    setFilteredList(filteredData);
+  }, [searchQuery, peopleData, sex, centuries.length]);
+
+  useEffect(() => {
```

Comment:
You are directly mutating the state by sorting the 'newSortedList'. Instead of directly sorting the state, make a copy and sort that to avoid unexpected behavior.